db/sqlc/migrations/000001_accounts.up.sql

```diff
@@ -0,0 +1,71 @@
+CREATE TABLE IF NOT EXISTS accounts (
+    -- The auto incrementing primary key.
+    id INTEGER PRIMARY KEY,
+
+    -- The ID that was used to identify the account in the legacy KVDB store.
+    -- In order to avoid breaking the API, we keep this field here so that
+    -- we can still look up accounts by this ID for the time being.
+    alias BIGINT NOT NULL UNIQUE,
+
+    -- An optional label to use for the account. If it is set, it must be
+    -- unique.
+    label TEXT UNIQUE,
+
+    -- The account type.
+    type SMALLINT NOT NULL,
+
+    -- The accounts initial balance. This is never updated.
+    initial_balance_msat BIGINT NOT NULL,
+
+    -- The accounts current balance. This is updated as the account is used.
+    current_balance_msat BIGINT NOT NULL,
+
+    -- The last time the account was updated.
+    last_updated TIMESTAMP NOT NULL,
+
+    -- The time that the account will expire.
+    expiration TIMESTAMP NOT NULL
+);
+
+-- The account_payments table stores all the payment hashes of outgoing
+-- payments that are associated with a particular account. These are used to
+-- when an account should be debited.
+CREATE TABLE IF NOT EXISTS account_payments (
+    -- The account that this payment is linked to.
+    account_id BIGINT NOT NULL REFERENCES accounts(id) ON DELETE CASCADE,
```

Comment:
It feels a bit dangerous to do `ON DELETE CASCADE` rather than `RESTRICT` here and for `account_invoices`, as in my opinion we should be really careful when deleting an account with existing payments and invoices, and really ensure that those have been deleted prior to deleting the account.

But given our mindset of not changing the original logic, and that's how it's handled today, I guess that motivates having it as a `CASCADE`.

Reply:
ooo TIL about RESTRICT! 

very nice

but yeah, let's keep as is for now to keep the tests passing :) can update during account revamp fosho